### PR TITLE
Faster implementation of dimension reduce for "flat" views.

### PR DESCRIPTION
**Pull request by martin-hora**

### CODE REVIEW

File: boltview/array_view.h [hunk 1]
```diff
@@ -148,12 +148,7 @@ class DeviceArrayView: public DeviceArrayConstView<TElement> {
 	DeviceArrayView &operator=(const DeviceArrayView &) = default;
 
 	BOLT_DECL_DEVICE
-	Element operator[](IndexType index) const {
```
Comment (martin-hora) [inline boltview/array_view.h:151]: I believe that this operator[] was unnecessary. Moreover, it caused problems during compilations, so I removed it.

**Comment by martin-hora** [comment]
ForEach accepts views indexed by int64_t (at least 1 dimensional views) without warning. I am afraid that it can lead to unexpected int overflows when computing grid sizes.

**Comment by martin-hora** [comment]
I added another commit that slightly improves the performance of reduction kernels on views with fundamental types (int, float, ...).
 It increases performance of dimension reduce kernel by 30 % (sum of floats in a device image view).
The reduce kernel is affected only for small bucket sizes. (About 10% boost for bucket size = 1).

**Comment by JanKolomaznik** [comment]
Looks good - merge.

**Pull request merged**